### PR TITLE
chore(docs,examples): snapshot docs and vanillajs example

### PR DESCRIPTION
**Pull request by RikuVan**
This is a first go at the snapshot docs, looks forward to corrections.

### CODE REVIEW

File: docs/advanced/snapshot.mdx [hunk 1]
```diff
@@ -1,7 +1,24 @@
...
 section: 'Advanced'
 description: 'Create a snapshot of current state'
 ---
 
 # snapshot
+
+#### A snapshot takes a proxy and returns an immutable object, unwrapped from the proxy. In sequential snapshot calls, when the values in the proxy have not changed, a pointer to the same object is returned. This allows for shallow comparison in render functions, preventing spurious renders. Snapshots also throw promises, making them work with React Suspense.
+
+```jsx
+import { proxy, snapshot } from 'valtio/vanilla'
```
Comment (dai-shi) [inline docs/advanced/snapshot.mdx:12]: Let's make this consistent with `subscribe`:
```suggestion
import { proxy, snapshot } from 'valtio'
```

And, later add a note that `valtio/vanilla` is available for vanilla-js users.

File: docs/advanced/snapshot.mdx [hunk 2]
```diff
@@ -1,7 +1,24 @@
 ---
-title: 'Snapshot'
+title: 'snapshot'
 section: 'Advanced'
 description: 'Create a snapshot of current state'
 ---
 
 # snapshot
+
+#### A snapshot takes a proxy and returns an immutable object, unwrapped from the proxy. In sequential snapshot calls, when the values in the proxy have not changed, a pointer to the same object is returned. This allows for shallow comparison in render functions, preventing spurious renders. Snapshots also throw promises, making them work with React Suspense.
```
Comment (dai-shi) [inline docs/advanced/snapshot.mdx:9]: We'd like to add a note that the immutable object is actually deeply frozen. Can be an additional note below.

File: examples/photo-booth-vanillajs/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "scripts": {
+    "dev": "vite",
+    "build": "vite build",
+    "start": "yarn run dev"
+  },
+  "dependencies": {
+    "lighterhtml": "4.2.0",
+    "localforage": "1.10.0",
+    "valtio": "1.3.0",
+    "vite": "2.8.1"
```
Comment (dai-shi) [inline examples/photo-booth-vanillajs/package.json:13]: Cool. It works with "githubbox" too. (and it's not that slow. one drawback is you can't modify code without login.)

**Review by dai-shi — COMMENTED** [review]
Thanks for working on this! Please see comments.

**Comment by dai-shi** [comment]
https://githubbox.com/RikuVan/valtio/tree/valtio-vanilla-docs/examples/photo-booth-vanillajs

image

wow, it's such a rich example.

File: docs/advanced/snapshot.mdx (hunk 1 above)
Comment (RikuVan) [reply docs/advanced/snapshot.mdx:12]: sure, it is important I think to note that you can import straight from valtio without react

**Review by dai-shi — APPROVED** [review]
LGTM
Thanks for your contribution!

**Pull request merged**